agent/pkg/podman/podman.go

```diff
@@ -0,0 +1,358 @@
+func (s *PodmanClient) RunContainer(
+	ctx context.Context,
+	waitCtx context.Context,
+	id string,
+	p events.Publisher[docker.Event],
+) (*docker.Container, error) {
+	cont := s.FindContainer(id)
+	if cont == nil {
+		return nil, container.ErrMissing
+	}
+	req := cont.Req
+
+	u, err := user.Current()
+	if err != nil {
+		return nil, fmt.Errorf("checking user: %w", err)
+	}
+
+	uidgid := fmt.Sprintf("%s:%s", u.Uid, u.Gid)
+	if req.ContainerConfig.User != uidgid {
+		return nil, fmt.Errorf(
+			"agent running as %s cannot launch as user %s",
+			uidgid, req.ContainerConfig.User,
+		)
+	}
+
+	tmpdir, err := os.MkdirTemp(agentTmp, fmt.Sprintf("*-%s", id))
+	if err != nil {
+		return nil, fmt.Errorf("making tmp dir for archives: %w", err)
+	}
+
+	var args []string
+	args = append(args, "run")
+	args = append(args, "--rm")
+	// args = append(args, "--writable-tmpfs")
+	args = append(args, "--workdir", req.ContainerConfig.WorkingDir)
+
+	// Env. variables. c.f. launcher PodmanOverSlurm.java
+	args = append(args, "--env", "SLURM_*")
+	args = append(args, "--env", "CUDA_VISIBLE_DEVICES")
+	args = append(args, "--env", "NVIDIA_VISIBLE_DEVICES")
+	args = append(args, "--env", "ROCR_VISIBLE_DEVICES")
+	args = append(args, "--env", "HIP_VISIBLE_DEVICES")
+
+	envFilePath := path.Join(tmpdir, envFileName)
+	envFile, err := os.OpenFile(
+		envFilePath,
+		os.O_APPEND|os.O_CREATE|os.O_WRONLY,
+		0o600,
+	) // #nosec G304 // We made this filepath, and it is randomized.
+	if err != nil {
+		return nil, fmt.Errorf("creating envfile %s: %w", envFilePath, err)
+	}
+	args = append(args, "--env-file", envFilePath)
+
+	req.ContainerConfig.Env = append(req.ContainerConfig.Env, "DET_NO_FLUENT=true")
+	req.ContainerConfig.Env = append(req.ContainerConfig.Env, "DET_SHIPPER_EMIT_STDOUT_LOGS=False")
+
+	for _, env := range req.ContainerConfig.Env {
+		_, err = envFile.WriteString(env + "\n")
+		if err != nil {
+			return nil, fmt.Errorf("writing to envfile: %w", err)
+		}
+	}
+
+	switch {
+	case req.HostConfig.NetworkMode == bridgeNetworking && s.Opts.AllowNetworkCreation:
+		// ?? publish ports only for bridgeNetworking
+		for port := range req.ContainerConfig.ExposedPorts {
+			p := port.Int()
+			args = append(args, "-p", fmt.Sprintf("%d:%d/tcp", p, p))
+		}
+		args = append(args, "--network=bridge")
+	case req.HostConfig.NetworkMode == bridgeNetworking:
+		if err = p.Publish(ctx, docker.NewLogEvent(
+			model.LogLevelDebug,
+			"container requested network virtualization, but network creation isn't allowed; "+
+				"overriding to host networking",
+		)); err != nil {
+			return nil, err
+		}
+		req.HostConfig.NetworkMode = hostNetworking
+		fallthrough
+	case req.HostConfig.NetworkMode == hostNetworking:
+		args = append(args, "--network=host")
+	default:
+		return nil, fmt.Errorf("unsupported network mode %s", req.HostConfig.NetworkMode)
+	}
+
+	archivesPath := filepath.Join(tmpdir, archivesName)
```

Comment:
Seems like the following section (archived expansion) should be a shared method of some time.  Only the CLI syntax is different, right?